src/lexer.c

```diff
@@ -418,10 +420,45 @@ void tryParseExportStatement () {
     facade = false;
     switch (ch) {
       // export default ...
-      case 'd':
-        addExport(pos, pos + 7, NULL, NULL);
+      case 'd': {
+        const char16_t* startPos = pos;
+        pos += 7;
+        ch = commentWhitespace(true);
+        // export default async? function*? name? (){}
+        if (ch == 'a' && memcmp(pos + 1, &SYNC[0], 4 * 2) == 0) {
+          pos += 5;
+          ch = commentWhitespace(true);
```

Comment:
I believe `async` and `function` need to be on the same line so that this should be `commentWhitespace(false)` strictly speaking. 

And this does actually have a real case it handles though believe it or not:

```js
const async = 2;
export default async

function x () {
}
```

Then in such a case, we need to ensure the fallthrough is handled properly as well instead of checking the next if statement.

Reply:
Thanks for the review and the great case, aha, I just learned that `async` is not a reserved word, I also handled the following case:
```
const asyncVar = 1
export default asyncVar
```